### PR TITLE
Capture server logs in tests

### PR DESCRIPTION
**Pull request by JamesNK**

### CODE REVIEW

**Comment by JamesNK** [comment]
Combined client and server logging:

```
2018-04-14 19:50:41.686 -07:00 [TestLifetime] [Information] Starting test "CanInvokeClientMethodFromServer_json_LongPolling_hubT"
2018-04-14 19:50:41.686 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Waiting on Connection Lock in "StartAsyncCore" ("C:\projects\signalr\src\Microsoft.AspNetCore.SignalR.Client.Core\HubConnection.cs":122).
2018-04-14 19:50:41.686 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Starting HubConnection.
2018-04-14 19:50:41.686 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.HttpConnection] [Debug] Starting HttpConnection.
2018-04-14 19:50:41.687 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.HttpConnection] [Debug] Establishing connection with server at 'http://127.0.0.1:3278/hubT'.
2018-04-14 19:50:41.687 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LoggingHttpMessageHandler] [Verbose] Sending HTTP request "POST" 'http://127.0.0.1:3278/hubT/negotiate'.
2018-04-14 19:50:41.687 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFM"" started.
2018-04-14 19:50:41.688 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request starting HTTP/1.1 POST http://127.0.0.1:3278/hubT/negotiate  0
2018-04-14 19:50:41.688 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] AuthenticationScheme: "Bearer" was not authenticated.
2018-04-14 19:50:41.688 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Request successfully matched the route with name 'null' and template '"/hubT/negotiate"'.
2018-04-14 19:50:41.688 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] New connection "ecKmsDUmbqg65evuN7BeNA" created.
2018-04-14 19:50:41.688 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Sending negotiation response.
2018-04-14 19:50:41.688 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFM"" completed keep alive response.
2018-04-14 19:50:41.688 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request finished in 0.7502ms 200 application/json
2018-04-14 19:50:41.688 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.HttpConnection] [Debug] Established connection '"ecKmsDUmbqg65evuN7BeNA"' with the server.
2018-04-14 19:50:41.689 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.HttpConnection] [Debug] Skipping transport "WebSockets" because it was disabled by the client.
2018-04-14 19:50:41.689 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.HttpConnection] [Debug] Skipping transport "ServerSentEvents" because it was disabled by the client.
2018-04-14 19:50:41.689 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.HttpConnection] [Debug] Starting transport '"LongPolling"' with Url: http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA.
2018-04-14 19:50:41.689 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Information] Starting transport. Transfer mode: Text.
2018-04-14 19:50:41.689 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Starting receive loop.
2018-04-14 19:50:41.689 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LoggingHttpMessageHandler] [Verbose] Sending HTTP request "GET" 'http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA'.
2018-04-14 19:50:41.689 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Starting the send loop.
2018-04-14 19:50:41.689 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.HttpConnection] [Debug] Transport 'LongPolling' started.
2018-04-14 19:50:41.689 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.HttpConnection] [Information] HttpConnection Started.
2018-04-14 19:50:41.689 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request starting HTTP/1.1 GET http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA  
2018-04-14 19:50:41.689 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] AuthenticationScheme: "Bearer" was not authenticated.
2018-04-14 19:50:41.690 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Request successfully matched the route with name 'null' and template '"/hubT"'.
2018-04-14 19:50:41.690 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Establishing new connection.
2018-04-14 19:50:41.690 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] OnConnectedAsync started.
2018-04-14 19:50:41.690 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Information] Using HubProtocol '"json" v1'.
2018-04-14 19:50:41.690 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Sending Hub Handshake.
2018-04-14 19:50:41.690 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Sending 32 bytes to the server using url: http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA.
2018-04-14 19:50:41.690 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LoggingHttpMessageHandler] [Verbose] Sending HTTP request "POST" 'http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA'.
2018-04-14 19:50:41.690 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFN"" started.
2018-04-14 19:50:41.691 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request starting HTTP/1.1 POST http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA  32
2018-04-14 19:50:41.691 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] AuthenticationScheme: "Bearer" was not authenticated.
2018-04-14 19:50:41.691 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Request successfully matched the route with name 'null' and template '"/hubT"'.
2018-04-14 19:50:41.691 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFN"", Request id ""0HLD2FEKS1IFN:00000001"": started reading request body.
2018-04-14 19:50:41.691 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFN"", Request id ""0HLD2FEKS1IFN:00000001"": done reading request body.
2018-04-14 19:50:41.691 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFN"" completed keep alive response.
2018-04-14 19:50:41.691 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request finished in 0.7765ms 200 text/plain
2018-04-14 19:50:41.691 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Found protocol implementation for requested protocol: "json".
2018-04-14 19:50:41.692 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Completed connection handshake. Using HubProtocol '"json"'.
2018-04-14 19:50:41.692 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Message(s) sent successfully.
2018-04-14 19:50:41.692 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFM"" completed keep alive response.
2018-04-14 19:50:41.692 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request finished in 2.7212ms 200 application/octet-stream
2018-04-14 19:50:41.692 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Verbose] Poll response with status code 200 received from server. Content length: 3.
2018-04-14 19:50:41.692 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Received messages from the server.
2018-04-14 19:50:41.692 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LoggingHttpMessageHandler] [Verbose] Sending HTTP request "GET" 'http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA'.
2018-04-14 19:50:41.692 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Handshake with server complete.
2018-04-14 19:50:41.692 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Receive loop starting.
2018-04-14 19:50:41.692 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Not using server timeout because the transport inherently tracks server availability.
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Information] HubConnection started.
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Releasing Connection Lock in "StartAsyncCore" ("C:\projects\signalr\src\Microsoft.AspNetCore.SignalR.Client.Core\HubConnection.cs":164).
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Registering handler for client method '"Echo"'.
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Waiting on Connection Lock in "InvokeCoreAsyncCore" ("C:\projects\signalr\src\Microsoft.AspNetCore.SignalR.Client.Core\HubConnection.cs":275).
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Client.InvocationRequest.NonStreaming] [Verbose] Invocation "1" created.
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Preparing blocking invocation '"1"' of '"CallEcho"', with return type '"System.Object"' and 1 argument(s).
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Registering Invocation ID '"1"' for tracking.
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Issuing Invocation '"1"': "System.Object" "CallEcho"("System.String").
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Sending "InvocationMessage" message '"1"'.
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Sending "InvocationMessage" message '"1"' completed.
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Releasing Connection Lock in "InvokeCoreAsyncCore" ("C:\projects\signalr\src\Microsoft.AspNetCore.SignalR.Client.Core\HubConnection.cs":288).
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request starting HTTP/1.1 GET http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA  
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] AuthenticationScheme: "Bearer" was not authenticated.
2018-04-14 19:50:41.693 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Request successfully matched the route with name 'null' and template '"/hubT"'.
2018-04-14 19:50:41.694 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Resuming existing connection.
2018-04-14 19:50:41.694 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Sending 74 bytes to the server using url: http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA.
2018-04-14 19:50:41.694 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LoggingHttpMessageHandler] [Verbose] Sending HTTP request "POST" 'http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA'.
2018-04-14 19:50:41.694 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request starting HTTP/1.1 POST http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA  74
2018-04-14 19:50:41.694 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] AuthenticationScheme: "Bearer" was not authenticated.
2018-04-14 19:50:41.694 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Request successfully matched the route with name 'null' and template '"/hubT"'.
2018-04-14 19:50:41.694 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFN"", Request id ""0HLD2FEKS1IFN:00000002"": started reading request body.
2018-04-14 19:50:41.695 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFN"", Request id ""0HLD2FEKS1IFN:00000002"": done reading request body.
2018-04-14 19:50:41.695 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFN"" completed keep alive response.
2018-04-14 19:50:41.695 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request finished in 1.0648ms 200 text/plain
2018-04-14 19:50:41.695 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Received hub invocation: "InvocationMessage { InvocationId: \"1\", Target: \"CallEcho\", Arguments: [ SignalR ] }".
2018-04-14 19:50:41.696 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Message(s) sent successfully.
2018-04-14 19:50:41.696 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFM"" completed keep alive response.
2018-04-14 19:50:41.696 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request finished in 2.6471ms 200 application/octet-stream
2018-04-14 19:50:41.696 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Verbose] Poll response with status code 200 received from server. Content length: 95.
2018-04-14 19:50:41.696 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Received messages from the server.
2018-04-14 19:50:41.696 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LoggingHttpMessageHandler] [Verbose] Sending HTTP request "GET" 'http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA'.
2018-04-14 19:50:41.696 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Processing 95 byte message from server.
2018-04-14 19:50:41.697 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Received Invocation 'null': "Echo"("System.String").
2018-04-14 19:50:41.697 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Received Completion for Invocation "1".
2018-04-14 19:50:41.697 -07:00 [Microsoft.AspNetCore.SignalR.Client.InvocationRequest.NonStreaming] [Verbose] Invocation "1" marked as completed.
2018-04-14 19:50:41.697 -07:00 [Microsoft.AspNetCore.SignalR.Client.InvocationRequest.NonStreaming] [Verbose] Invocation "1" disposed.
2018-04-14 19:50:41.697 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Waiting on Connection Lock in "StopAsyncCore" ("C:\projects\signalr\src\Microsoft.AspNetCore.SignalR.Client.Core\HubConnection.cs":180).
2018-04-14 19:50:41.697 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Releasing Connection Lock in "StopAsyncCore" ("C:\projects\signalr\src\Microsoft.AspNetCore.SignalR.Client.Core\HubConnection.cs":207).
2018-04-14 19:50:41.697 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Stopping HubConnection.
2018-04-14 19:50:41.697 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Terminating receive loop.
2018-04-14 19:50:41.697 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Waiting for the receive loop to terminate.
2018-04-14 19:50:41.697 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Waiting on Connection Lock in "ReceiveLoop" ("C:\projects\signalr\src\Microsoft.AspNetCore.SignalR.Client.Core\HubConnection.cs":652).
2018-04-14 19:50:41.698 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Releasing Connection Lock in "ReceiveLoop" ("C:\projects\signalr\src\Microsoft.AspNetCore.SignalR.Client.Core\HubConnection.cs":661).
2018-04-14 19:50:41.698 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.HttpConnection] [Debug] Disposing HttpConnection.
2018-04-14 19:50:41.698 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Information] Transport is stopping.
2018-04-14 19:50:41.698 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Send loop canceled.
2018-04-14 19:50:41.698 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Send loop stopped.
2018-04-14 19:50:41.699 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Receive loop stopped.
2018-04-14 19:50:41.699 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Sending DELETE request to 'http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA'.
2018-04-14 19:50:41.699 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LoggingHttpMessageHandler] [Verbose] Sending HTTP request "DELETE" 'http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA'.
2018-04-14 19:50:41.699 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request starting HTTP/1.1 GET http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA  
2018-04-14 19:50:41.699 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] AuthenticationScheme: "Bearer" was not authenticated.
2018-04-14 19:50:41.700 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Request successfully matched the route with name 'null' and template '"/hubT"'.
2018-04-14 19:50:41.700 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Resuming existing connection.
2018-04-14 19:50:41.700 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFM"" received FIN.
2018-04-14 19:50:41.700 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFM"" disconnecting.
2018-04-14 19:50:41.700 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request starting HTTP/1.1 DELETE http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA  0
2018-04-14 19:50:41.700 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] AuthenticationScheme: "Bearer" was not authenticated.
2018-04-14 19:50:41.700 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Request successfully matched the route with name 'null' and template '"/hubT"'.
2018-04-14 19:50:41.701 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFN"" completed keep alive response.
2018-04-14 19:50:41.701 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Information] Request finished in 0.5676ms 202 text/plain
2018-04-14 19:50:41.701 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] Connection id ""0HLD2FEKS1IFM"" sending FIN.
2018-04-14 19:50:41.701 -07:00 [Microsoft.AspNetCore.SignalR.Tests.ServerLogScope] [Debug] OnConnectedAsync ending.
2018-04-14 19:50:41.701 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] DELETE request to 'http://127.0.0.1:3278/hubT?id=ecKmsDUmbqg65evuN7BeNA' accepted.
2018-04-14 19:50:41.701 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport] [Debug] Transport stopped.
2018-04-14 19:50:41.701 -07:00 [Microsoft.AspNetCore.Http.Connections.Client.HttpConnection] [Information] HttpConnection Disposed.
2018-04-14 19:50:41.702 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] Canceling all outstanding invocations.
2018-04-14 19:50:41.702 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Verbose] Shutting down connection.
2018-04-14 19:50:41.702 -07:00 [Microsoft.AspNetCore.SignalR.Client.HubConnection] [Debug] HubConnection stopped.
2018-04-14 19:50:41.702 -07:00 [TestLifetime] [Information] Finished test "CanInvokeClientMethodFromServer_json_LongPolling_hubT" in 0.016217s
```

File: src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/WebSocketsTransport.cs [hunk 1]
```diff
@@ -182,6 +182,14 @@ public WebSocketsTransport(WebSocketOptions options, IDuplexPipe application, Ht
                     }
                 }
             }
+            catch (WebSocketException ex) when (
+                ex.ErrorCode == 997 ||
+                // Sometimes this error is raised without the ErrorCode
+                ex.Message == "The remote party closed the WebSocket connection without completing the close handshake.")
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/WebSocketsTransport.cs]: There's a specific websocket error code to look for.

Comment (JamesNK) [reply src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/WebSocketsTransport.cs]: I know, 997, the line above. I found that sometimes it isn't set

Comment (davidfowl) [reply src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/WebSocketsTransport.cs]: There's another property called WebsocketErrorCode

Comment (JamesNK) [reply src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/WebSocketsTransport.cs]: That is also not set

Comment (JamesNK) [reply src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/WebSocketsTransport.cs]: People getting this issue: https://github.com/aspnet/AspNetCoreModule/issues/77

Comment (davidfowl) [reply src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/WebSocketsTransport.cs]: I don’t want this text comparison in our code. We control the websocket implementation. If it doesn’t set the right flag, we should fix it. It looks like it does so let’s use that.

Comment (JamesNK) [reply src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/WebSocketsTransport.cs]: Should the error be ignored at all? Are WebSocketsTest and WebSocketsReceivesAndSendsPartialFramesTest written correctly? Or is it a bug in the framework? https://github.com/aspnet/AspNetCoreModule/issues/77

Comment (davidfowl) [reply src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/WebSocketsTransport.cs]: Yes it should be ignored. It happens for example if an desktop application closes. We basically want to ignore errors that can happen if a client disconnects

**Comment by davidfowl** [comment]
@JamesNK I think this change is done but I don't want to merge it until we make the CI greener than it currently is.

**Comment by anurse** [comment]
Why does this work when our previous attempts failed? We run tests in parallel, and they all share the server fixture. Has that changed? Am I missing something?

**Comment by anurse** [comment]
Gotcha. I don't mind ignoring the problem as long as we all realize how it works :). It's still useful to have I suppose.

The big problem is going to be remembering this when diagnosing flaky tests on the CI using the file logs. The server logs are going to be confusing AF.

**Comment by davidfowl** [comment]
> The big problem is going to be remembering this when diagnosing flaky tests on the CI using the file logs. The server logs are going to be confusing AF.

I don't disagree. @JamesNK do you have any ideas on how we can fix it?

**Comment by JamesNK** [comment]
I'm not sure the server tests are run in parallel. Whenever I look at the logs it seems like they are only showing server requests for the current tests.

**Comment by JamesNK** [comment]
You can add `DisableParallelization=true` against a collection to prevent tests running in parallel.

**Comment by davidfowl** [comment]
@JamesNK how much slower do the tests run?

**Comment by JamesNK** [comment]
The same. I think they already weren't being run in parallel. Maybe it is disabled when the tests are invoked from the command line? idk

**Comment by JamesNK** [comment]
Now that the long polling bug is fix, this should be more stable.

**Comment by davidfowl** [comment]
I want this in release, it'll help diagnose issues we hit when we're patching @muratg

**Pull request merged**